cmd/source_profile.go

```diff
@@ -84,26 +103,42 @@ type SourceProfileConnectionPostgreSQL struct {
 	pwd  string // Same as PGPASSWORD environment variable
 }
 
-func NewSourceProfileConnectionPostgreSQL(params map[string]string) SourceProfileConnectionPostgreSQL {
+func NewSourceProfileConnectionPostgreSQL(params map[string]string) (SourceProfileConnectionPostgreSQL, error) {
 	pg := SourceProfileConnectionPostgreSQL{}
-	if host, ok := params["host"]; ok {
-		pg.host = host
-	}
-	if port, ok := params["port"]; ok {
-		pg.port = port
-	} else { // Set default port for postgresql, which rarely changes.
-		pg.port = "5432"
+	host, hostOk := params["host"]
+	user, userOk := params["user"]
+	db, dbOk := params["db_name"]
+	port, portOk := params["port"]
+	pwd, pwdOk := params["password"]
+	if hostOk && userOk && dbOk {
+		// All connection params provided through source-profile. Port and password handled later.
+		pg.host, pg.user, pg.db, pg.port, pg.pwd = host, user, db, port, pwd
+	} else if !hostOk && !userOk && !dbOk && !portOk && !pwdOk {
+		// No connection params provided through source-profile. Fetching from env variables.
+		fmt.Printf("Connection parameters not specified in source-profile. Reading from environment variables...\n")
+		pg.host = os.Getenv("PGHOST")
+		pg.user = os.Getenv("PGUSER")
+		pg.db = os.Getenv("PGDATABASE")
+		pg.port = os.Getenv("PGPORT")
+		pg.pwd = os.Getenv("PGPASSWORD")
+	} else {
+		return pg, fmt.Errorf("please specify host, port, user and db_name in the source profile")
```

Comment:
s\source profile\source-profile (here and elsewhere in case I missed this in my PRs)